openpype/plugins/publish/extract_color_transcode.py

```diff
@@ -0,0 +1,333 @@
+
+            for output_name, output_def in profile.get("outputs", {}).items():
+                new_repre = copy.deepcopy(repre)
+
+                original_staging_dir = new_repre["stagingDir"]
+                new_staging_dir = get_transcode_temp_directory()
+                new_repre["stagingDir"] = new_staging_dir
+
+                if isinstance(new_repre["files"], list):
+                    files_to_convert = copy.deepcopy(new_repre["files"])
+                else:
+                    files_to_convert = [new_repre["files"]]
+
+                output_extension = output_def["extension"]
+                output_extension = output_extension.replace('.', '')
+                self._rename_in_representation(new_repre,
+                                               files_to_convert,
+                                               output_name,
+                                               output_extension)
+
+                target_colorspace = output_def["colorspace"]
+                view = output_def["view"] or colorspace_data.get("view")
+                display = (output_def["display"] or
+                           colorspace_data.get("display"))
+                # both could be already collected by DCC,
+                # but could be overwritten
+                if view:
+                    new_repre["colorspaceData"]["view"] = view
+                if display:
+                    new_repre["colorspaceData"]["display"] = display
+
+                files_to_convert = self._translate_to_sequence(
+                    files_to_convert)
+                for file_name in files_to_convert:
+                    input_path = os.path.join(original_staging_dir,
+                                              file_name)
+                    output_path = self._get_output_file_path(input_path,
+                                                             new_staging_dir,
+                                                             output_extension)
+                    convert_colorspace(
+                        input_path,
+                        output_path,
+                        config_path,
+                        source_colorspace,
+                        target_colorspace,
+                        view,
+                        display,
+                        self.log
+                    )
+
+                # cleanup temporary transcoded files
+                for file_name in new_repre["files"]:
+                    transcoded_file_path = os.path.join(new_staging_dir,
+                                                        file_name)
+                    instance.context.data["cleanupFullPaths"].append(
+                        transcoded_file_path)
+
+                custom_tags = output_def.get("custom_tags")
+                if custom_tags:
+                    if not new_repre.get("custom_tags"):
+                        new_repre["custom_tags"] = []
+                    new_repre["custom_tags"].extend(custom_tags)
```

Comment:
```suggestion
                custom_tags = output_def.get("custom_tags")
                if custom_tags:
                    if "custom_tags" not in new_repre.get:
                        new_repre["custom_tags"] = []
                    new_repre["custom_tags"].extend(custom_tags)
```

Reply:
This syntax is just weird @_@. Any special reason why this is better?

Reply:
Because `if not repre.get("custom_tags")` is checking output value, not existence of key.
```
repre = {"custom_tags": []}
# Even if the key is there and is set it will be set again
if not repre.get("custom_tags"):
    repre["custom_tags"] = []
```

It could be also changed to
```
if repre.get("custom_tags") is None:
    repre["custom_tags"] = []
```